.github/workflows/ci.yml

```diff
@@ -2,6 +2,8 @@ name: CI
 
 on:
   push:
+    branches:
+      - main
```

Comment:
What will `${{ env.TAG }}` be on main?

Reply:
|Scenario|Trigger Type|$TAG
|---------|--------------|-----
|Push to main|push|main|
|Tag push (e.g., v1.2.3)|push|v1.2.3|
|PR from feature-x to main|pull_request|feature-x|

we can add an `if` in this step to make sure it only runs on Tag push as well